### PR TITLE
[FIX] web_editor: fix the resizing history

### PR DESCRIPTION
**Pull request by sobo-odoo**
This commit fixes the issue where the undo of the resizing of an
element is not done in one step.

Steps to reproduce:
- drop a snippet (like text-image)
- resize the image
- undo
=> you have to click multiple times before going back to the initial
size.

task-2968749

### CODE REVIEW

**Comment by sobo-odoo** [comment]
@qsm-odoo It's ready for review :)

File: addons/web_editor/static/src/js/editor/snippets.options.js [hunk 1]
```diff
@@ -4133,6 +4133,7 @@ registry.sizing = SnippetOptionWidget.extend({
         var resizeValues = this._getSize();
         this.$handles.on('mousedown', function (ev) {
             ev.preventDefault();
+            self.options.wysiwyg.odooEditor.automaticStepUnactive();
```
Comment (qsm-odoo) [inline addons/web_editor/static/src/js/editor/snippets.options.js:4136]: Mmh, I am not sure about this (probably right but not enough). If it is enough, it lacks comments IMO. The description in your commit message actually points to the fact that something is very strange on its own: why is the resizing influenced by the fact you moved an element first (the solution probably cannot be just about the resizing code on its own). I think the drag and drop code is wrong:

- Drag and drop of something in the page:

`_onDragAndDropStart`:
```javascript
this.options.wysiwyg.odooEditor.automaticStepUnactive();
```

`_onDragAndDropStop`:
```javascript
this.options.wysiwyg.odooEditor.automaticStepActive();
this.options.wysiwyg.odooEditor.automaticStepSkipStack();
this.options.wysiwyg.odooEditor.unbreakableStepUnactive();
```

- Drag and drop of a snippet from the editor panel:

`start`:
```javascript
self.options.wysiwyg.odooEditor.automaticStepUnactive();
```

`stop`:
```javascript
self.options.wysiwyg.odooEditor.automaticStepUnactive();
self.options.wysiwyg.odooEditor.automaticStepSkipStack();
```

So basically, for both drag and drop, the start is the same but the the stop is very different. I can't guess why and there is no comment around those lines and there is no documentation for those methods... maybe @Zinston can help us so we can make a more complete fix? Unless you have an explanation for this @sobo-odoo ? :thinking:

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/web_editor/static/src/js/editor/snippets.options.js (hunk 1 above)
Comment (Goaman) [reply addons/web_editor/static/src/js/editor/snippets.options.js:4136]: I've made a PR to fix the `stop` problem in 15.0: https://github.com/odoo/odoo/pull/101167

Comment (sobo-odoo) [reply addons/web_editor/static/src/js/editor/snippets.options.js:4136]: I know the goal was not to solve the resizing history problem but just to inform that now it happens everytime (with your fix). So can we say that this is in fact the normal behavior and that the problem really was something missing in the resizing part ?  And if I understood correctly, I should also add an `automaticStepActive` after the resizing is done ?

Comment (Goaman) [reply addons/web_editor/static/src/js/editor/snippets.options.js:4136]: > So can we say that this is in fact the normal behavior and that the problem really is something missing in the resizing part

The fact that resizing takes multiples history step is not expected and should be fixed.

The comment of @qsm-odoo made me investigate our code about the `stop` of the dnd, which led to my fix.
Indeed, the fix is not directly related and do not solve this resizing taking multiples steps.

> And if I understood correctly, I should also add an automaticStepActive after the resizing is done is that right ?

Yes. If it does not work, we can have a look together.

**Comment by sobo-odoo** [comment]
@qsm-odoo I made the changes :)

**Review by qsm-odoo — APPROVED** [review]
@robodoo rebase-ff r+ 

(I included NBY's fix here directly (https://github.com/odoo/odoo/pull/101167))
